### PR TITLE
Ft user filters

### PR DESCRIPTION
**Pull request by bizip**
#### What does this PR do?

Add filters to the list of users

#### Description of Task to be completed?

- [x] Filter the user's list by inputting text

- [x]  Filter the user's list by role

- [x] Filter the user's list by projects

#### How should this be manually tested?
After cloning the repo, cd into it, and run `yarn  install && yarn run dev`

- Using any browser head to localhost:3000, sign in with Microsoft account

-  Click on the user's section from the admin dashboard

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker/Trello stories?
N/A
#### Screenshots (if appropriate)
image

### CODE REVIEW

File: src/interfaces/user.interface.ts [hunk 1]
```diff
@@ -6,4 +6,5 @@ export interface User {
   role: Role;
   email: string;
   isActivated: boolean;
+  users: [];
```
Comment (nicele08) [inline src/interfaces/user.interface.ts]: @bizip give it type like `users: User[]`

File: src/redux/features/users/userSlice.ts [hunk 2]
```diff
@@ -18,16 +18,52 @@ export const getAllUsers = createAsyncThunk(
...
+    // eslint-disable-next-line no-useless-catch
+    try{
+      const { data } = await API.get('/projects');
+      return data;
+    }catch(error: any){
+      throw error?.response?.data;
+    }
+   
+  },
+);
```
Comment (nicele08) [inline src/redux/features/users/userSlice.ts:34]: @bizip why do we have this in userSlice, I think @Cherubln worked on fetching projects so I recommunicate with him so we don't duplicate the same knowledge.

File: src/redux/features/users/userSlice.ts [hunk 3]
```diff
@@ -18,16 +18,52 @@ export const getAllUsers = createAsyncThunk(
...
+
 interface InitialState {
   users: User[];
+  usersByProject: User[];
+  projects:any [],
```
Comment (nicele08) [inline src/redux/features/users/userSlice.ts]: @bizip Avoid using `any` type wherever possible, so define the project type and use it here instead

**Review by nicele08 — CHANGES_REQUESTED** [review]
@bizip check the comments and try to work them

File: src/redux/features/users/userSlice.ts (hunk 2 above)
Comment (bizip) [reply src/redux/features/users/userSlice.ts:34]: Thank you @nicele08 but I have seen that there were no get all the projects therefore I have used that one.

**Pull request merged**